lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -300,21 +315,49 @@ def _on_scrape_targets_changed(self, event):
 logger = logging.getLogger(__name__)
 
 
+JOB_KEYS = {
```

Comment:
Is it necessary to be an intermediary? Could just pass them all as they are (and remove this list entirely).
Currently you do check validity of duration,but there are many other things that would make a configuration invalid.

Reply:
I would prefer keeping the list of allowed job keys as a global constant. This is and important data structure for the entire library. This also allows reuse if required and keeps my santization function small. Prometheus charm does do validation.